components/cam/src/physics/cam/micro_p3.F90

```diff
@@ -405,10 +406,14 @@ SUBROUTINE p3_main_part1(kts, kte, kbot, ktop, kdir, do_predict_nc, dt, &
       !--- Apply droplet activation here (before other microphysical processes) for consistency with qc increase by saturation
       !    adjustment already applied in macrophysics. If prescribed drop number is used, this is also a good place to
       !    prescribe that value
-          if (.not.(do_predict_nc)) then
+          if (.not.(do_predict_nc) .and. .not.(do_prescribed_CCN)) then
             nc(k) = nccnst*inv_rho(k)
           else
-            nc(k) = max(nc(k) + nc_nuceat_tend(k) * dt,0.0_rtype)
+            if (.not. (do_prescribed_CCN)) then
+               nc(k) = max(nc(k) + nc_nuceat_tend(k) * dt,0.0_rtype)
+            else
+               nc(k) = max(nc(k),nccn_prescribed(k))
+            endif
```

Comment:
less contorted logic here would be "if do_prescribed_CCN...else if do_predict_nc... else <do not do_predict_nc and not do_prescribed_ccn>"

Reply:
fixed.